src/ring_theory/polynomial/hermite.lean

```diff
@@ -124,5 +129,57 @@ begin
 end
 
 end coeff
+end polynomial
+
+/-! ### Lemmas about `polynomial.hermite_gauss` -/
+
+section gaussian
+
+/-- The real Gaussian function -/
+def gaussian : ℝ → ℝ := λ x, real.exp (-(x^2 / 2))
+
+lemma inv_gaussian_eq : gaussian⁻¹ = λ x, real.exp (x^2 / 2) :=
+by { ext, simp [gaussian, real.exp_neg] }
+
+lemma inv_gaussian_mul_gaussian (x : ℝ) : gaussian⁻¹ x * gaussian x = 1 :=
+inv_mul_cancel (real.exp_pos _).ne.symm
+
+lemma deriv_gaussian (x : ℝ) : deriv gaussian x = -x * gaussian x :=
+by simp [gaussian, mul_comm]
+
+lemma deriv_inv_gaussian (x : ℝ) : deriv gaussian⁻¹ x = x * gaussian⁻¹ x :=
+by simp [inv_gaussian_eq, mul_comm]
+
+lemma cont_diff_gaussian : cont_diff ℝ ⊤ gaussian :=
+((cont_diff_id.pow 2).div_const 2).neg.exp
+
+namespace polynomial
+
+lemma hermite_gauss_succ (n : ℕ) :
+(λ x, (-1)^(n+1) * (gaussian⁻¹ x) * (deriv^[n+1] gaussian x))
+= id * (λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) -
+  deriv (λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) :=
+begin
+  ext,
+  simp only [function.iterate_succ', function.comp_app,
+             id.def, pi.mul_apply, pi.sub_apply, pow_succ],
+  rw [deriv_mul, deriv_const_mul, deriv_inv_gaussian],
+  ring,
+  { simp [inv_gaussian_eq] },
+  { simp [inv_gaussian_eq] },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ cont_diff_gaussian)).1 }
+end
+
+lemma hermite_eq_deriv_gaussian (n : ℕ) (x : ℝ) :
+  aeval x (hermite n) = (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) :=
+begin
+  apply function.funext_iff.mp _ x,
+  induction n with n ih,
```

Comment:
I would guess you can remove the line above if you change this to
```suggestion
  induction n with n ih generalizing x,
```

Reply:
`generalizing x` keeps `x` in context and doesn't produce a lambda expression (it produces a `∀ (x : ℝ), ...` goal and inductive hypothesis). Then `rw hermite_gauss_succ` doesn't work (and neither does `rw congr_fun (hermite_gauss_succ n) x` because of annoying stuff about `dsimp`. Maybe there's a way to do this, but I couldn't find a way without extra steps elsewhere.

Reply:
`simp_rw` should work in the inductive hypothesis

Reply:
But `hermite_gauss_succ` is a pretty weirdly stated lemma too; I'd encourage you to remove the lambdas there too, or even just to inline it into this proof

Reply:
Okay, this works with appropriate modifications to `hermite_gauss_succ`.
```
lemma hermite_gauss_succ (n : ℕ) (x : ℝ) :
(-1)^(n + 1) * (gaussian⁻¹ x) * (deriv^[n + 1] gaussian x)
= x * ((-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) -
  deriv (λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) x :=
begin
  simp only [function.iterate_succ', function.comp_app,
             id.def, pi.mul_apply, pi.sub_apply],
  rw [deriv_mul, deriv_const_mul, deriv_inv_gaussian],
  ring_exp,
  { simp [inv_gaussian_eq] },
  { simp [inv_gaussian_eq] },
  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ cont_diff_gaussian)).1 }
end

lemma hermite_eq_deriv_gaussian (n : ℕ) (x : ℝ) :
  aeval x (hermite n) = (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) :=
begin
  induction n with n ih generalizing x,
  { simp [-pi.inv_apply, inv_gaussian_mul_gaussian] },
  { simp_rw [hermite_gauss_succ, hermite_succ, ← ih],
    simp [aeval_def, eval₂_eq_eval_map] },
end
```

Reply:
Okay, this works with appropriate modifications to `hermite_gauss_succ`.
```
lemma hermite_gauss_succ (n : ℕ) (x : ℝ) :
(-1)^(n + 1) * (gaussian⁻¹ x) * (deriv^[n + 1] gaussian x)
= x * ((-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) -
  deriv (λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) x :=
begin
  simp only [function.iterate_succ', function.comp_app,
             id.def, pi.mul_apply, pi.sub_apply],
  rw [deriv_mul, deriv_const_mul, deriv_inv_gaussian],
  ring_exp,
  { simp [inv_gaussian_eq] },
  { simp [inv_gaussian_eq] },
  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ cont_diff_gaussian)).1 }
end

lemma hermite_eq_deriv_gaussian (n : ℕ) (x : ℝ) :
  aeval x (hermite n) = (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) :=
begin
  induction n with n ih generalizing x,
  { simp [-pi.inv_apply, inv_gaussian_mul_gaussian] },
  { simp_rw [hermite_gauss_succ, hermite_succ, ← ih],
    simp [aeval_def, eval₂_eq_eval_map] },
end
```